skopt/local_solve/paramils.py

```diff
@@ -0,0 +1,109 @@
+import numpy as np
+
+from sklearn.externals.joblib import Parallel, delayed
+from sklearn.utils import check_random_state
+
+def _transform_back(x0, lower_bounds, upper_bounds):
+    return x0 * (upper_bounds - lower_bounds) + lower_bounds
+
+def _local_search(func, x0, bounds, std, n_neighbors,
+                  random_state=None, maxiter=10):
+    rng = check_random_state(random_state)
+    upper_bounds = bounds[:, 1]
+    lower_bounds = bounds[:, 0]
+
+    transformed_x_min = np.asarray(x0)
+    actual_x_min = _transform_back(
+        transformed_x_min, lower_bounds, upper_bounds)
+
+    fun_min = func(actual_x_min.reshape(1, -1))[0]
+    n_params = len(x0)
+
+    for i in range(maxiter):
+        # TODO: Cache and modify in-place.
+        transformed_points = np.tile(
+            transformed_x_min, (n_neighbors * n_params, 1))
+
+        for n_param in range(n_params):
```

Comment:
We should really try to be consistent across the whole package about how we name things...